### ISSUE TITLE
Plotting DBSCAN clustering results of a leveinshtein similarity matrix

### ISSUE THREAD

**Issue by laminenoureddine** [EpistasisLab/tpot#844]
Hello,

 I have a leveinshtein similarity matrix with 13600x13600 samples. I used DBSCAN to cluster this dataset. I got results, but I want to plot these clusters.

 After some googling, I see that this problem could be solved by firstly getting the coordinates of the similarity matrix I have, and then when done, just plot the coordinates with their labels (each unique label is coloured differently of course).

 Most of answers recommend to use MDS for such situation. So, I test it, but it seems that it's very very costly (the process tends to terminate with "Memory error").

 So, my questions/discussions are:

1 - Is there another way/approach to plot the clustering results of my leveinshtein similarity matrix, rather then getting first their coordinates then plot them in a 2D plan with their coloured clustering results labels ?
2 - Instead of MDS, Is there another algorithm to do the job of finding coordinates of a similarity matrix  ? I'm aware that PCA works on reducing the dimensionality of the features, but in my case I've only the similarity matrix, not the features.
3 - Is there an optmized version of MDS for my 13kx13k dataset ?
4 - Also, is MDS working only on a euclidean distance matrix ? I mean is it working properly for my leveinshtein distance matrix ?

Thanks in advance for your help,

**Comment by weixuanfu** [comment]
Hi, is this a TPOT related question? I think it is more related to [DBSCAN](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.DBSCAN.html). Could you please post this issue to scikit-learn's repo?

**Comment by weixuanfu** [comment]
Please feel free to reopen this issue if it is a TPOT related issue/question.
